### PR TITLE
Eng 131 update webhook conversation api

### PR DESCRIPTION
**Pull request by Ali-BenZarrouk-MB**
https://messagebird.atlassian.net/browse/ENG-131

### CODE REVIEW

File: api/src/main/java/com/messagebird/MessageBirdClient.java [hunk 1]
```diff
@@ -855,12 +855,27 @@ public void deleteConversationWebhook(final String webhookId)
      * @param request Webhook to create.
      * @return Newly created webhook.
      */
-    public ConversationWebhook sendConversationWebhook(final ConversationWebhookRequest request)
+    public ConversationWebhook sendConversationWebhook(final ConversationWebhookCreateRequest request)
```
Comment (epels) [inline api/src/main/java/com/messagebird/MessageBirdClient.java]: Please revert this change as this breaks backwards compatibility

File: api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package com.messagebird.objects.conversations;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Contains common fields for webhook requests.
+ */
+public abstract class ConversationWebhookBaseRequest {
```
Comment (epels) [inline api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java]: Although the idea is nice and it works well now, I'm not a fan of having a shared base class for the create and update endpoints. They're separate endpoints, so imho they should be completely separate in the SDK as well. If the API changes in the future, this would also force us to publish a new major version of this SDK. What do you think?

File: examples/pom.xml [hunk 3]
```diff
@@ -20,7 +20,8 @@
         <dependency>
             <groupId>com.messagebird</groupId>
             <artifactId>messagebird-api</artifactId>
-            <version>2.1.1</version>
+            <version>2.1.2</version>
```
[the remark this answers is not in the document]
Comment (epels) [reply examples/pom.xml]: And here: https://github.com/messagebird/java-rest-api/blob/master/api/src/main/java/com/messagebird/MessageBirdServiceImpl.java#L60

File: api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    public List<ConversationWebhookEvent> getEvents() {
+        return events;
+    }
+
+    public void setEvents(List<ConversationWebhookEvent> events) {
+        this.events = events;
+    }
+
+    protected abstract String getRequestName();
```
Comment (epels) [inline api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java]: This is not used anywhere, is it? In that case I propose removing it

File: api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java (hunk 2 above)
Comment (denizkilic) [reply api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java]: For me, using a base class for creating and updating webhook endpoints makes sense in here because those endpoints use the same domain object.

File: api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java (hunk 4 above)
Comment (Ali-BenZarrouk-MB) [reply api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java]: It is used in the ToString of the base.

File: examples/src/main/java/ExampleUpdateConversationWebhook.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import com.messagebird.objects.conversations.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExampleUpdateConversationWebhook {
+
+    public static void main(String[] args) {
+        if (args.length < 4 || args.length > 8) {
+            System.out.println("Please specify your access key, webhook id, enabled or not, url of webhook and events.\n" +
```
[the remark this answers is not in the document]
Comment (Ali-BenZarrouk-MB) [reply examples/src/main/java/ExampleUpdateConversationWebhook.java]: I will add support for default values. And we can keep the possibilities for running with custom values.

File: api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java (hunk 4 above)
Comment (epels) [reply api/src/main/java/com/messagebird/objects/conversations/ConversationWebhookBaseRequest.java]: Ah ok, missed that 👍
